hyperspy/_signals/hologram_image.py

```diff
@@ -0,0 +1,392 @@
+            Sideband radius of the aperture in corresponding unit (see 'sb_unit'). If None,
+            the radius of the aperture is set to 1/3 of the distance between sideband and
+            center band.
+        sb_smoothness : float, ndarray, :class:`~hyperspy.signals.BaseSignal, None
+            Smoothness of the aperture in the same unit as sb_size.
+        sb_unit : str, None
+            Unit of the two sideband parameters 'sb_size' and 'sb_smoothness'.
+            Default: None - Sideband size given in pixels
+            'nm': Size and smoothness of the aperture are given in 1/nm.
+            'mrad': Size and smoothness of the aperture are given in mrad.
+        sb : str, None
+            Select which sideband is selected. 'upper' or 'lower'.
+        sb_position : tuple, :class:`~hyperspy.signals.Signal1D, None
+            The sideband position (y, x), referred to the non-shifted FFT. If None, sideband is determined
+            automatically from FFT.
+        output_shape: tuple, None
+            Choose a new output shape. Default is the shape of the input hologram. The output
+            shape should not be larger than the input shape.
+        plotting : boolean
+            Shows details of the reconstruction (i.e. SB selection).
+        show_progressbar : boolean
+            Shows progressbar while iterating over different slices of the signal (passes the parameter to map method).
+        store_parameters : boolean
+            Store reconstruction parameters in metadata
+
+        Returns
+        -------
+        wave : :class:`~hyperspy.signals.WaveImage
+            Reconstructed electron wave. By default object wave is devided by reference wave
+
+        Notes
+        -----
+
+        """
+
+        # TODO: Use defaults for choosing sideband, smoothness, relative filter size and output shape if not provided
+        # TODO: Plot FFT with marked SB and SB filter if plotting is enabled
+
+        # Parsing reference:
+        if not isinstance(reference, HologramImage):
+            if isinstance(reference, Signal2D):
+                if (not reference.axes_manager.navigation_shape == self.axes_manager.navigation_shape
+                    and reference.axes_manager.navigation_size):
+
+                    raise ValueError('The navigation dimensions of object and reference holograms do not match')
+
+                _logger.warning('The reference image signal type is not HologramImage. It will '
+                                'be converted to HologramImage automatically.')
+                reference.set_signal_type('hologram')
+            elif reference is not None:
+                reference = HologramImage(reference)
+
+        # Testing match of navigation axes of reference and self (exception: reference nav_dim=1):
+        if (reference and not reference.axes_manager.navigation_shape == self.axes_manager.navigation_shape
+            and reference.axes_manager.navigation_size):
```

Comment:
Shouldn't we also check if the signal shape of the reference image matches the one of `self`?